fem/fespace.cpp

```diff
@@ -503,6 +503,31 @@ void FiniteElementSpace::BuildDofToArrays()
    }
 }
 
+void FiniteElementSpace::BuildDofToBdrArrays() const
+{
+   if (dof_bdr_elem_array.Size()) { return; }
+
+   BuildBdrElementToDofTable();
+
+   dof_bdr_elem_array.SetSize (ndofs);
+   dof_bdr_ldof_array.SetSize (ndofs);
+   dof_bdr_elem_array = -1;
+   for (int i = 0; i < mesh -> GetNBE(); i++)
+   {
+      const int *dofs = bdr_elem_dof -> GetRow(i);
+      const int n = bdr_elem_dof -> RowSize(i);
+      for (int j = 0; j < n; j++)
+      {
+         int dof = DecodeDof(dofs[j]);
+         if (dof_bdr_elem_array[dof] < 0)
```

Comment:
Here, check if `dof_bdr_elem_array[dof] < 0` and then overwrite it too.

Reply:
Sorry I'm not sure what you mean here -- this conditional is to make sure that for a given bdr dof, only the first boundary element index `i` when looping from 0 to `GetNBE()` for that dof is associated with it.

Reply:
Ahh, right, that is a nonsense. I thought the loop goes differently.